pandas/core/dtypes/common.py

```diff
@@ -220,6 +220,10 @@ def is_categorical(arr):
     return isinstance(arr, ABCCategorical) or is_categorical_dtype(arr)
 
 
+def is_range(arr):
```

Comment:
you are conflating a type with an Index. is_categorical will detect a categorical type which could be a Categorical (or the dtype=='category'), a CategoricalIndex happens to be of this type as well.

However RangeIndex is simply an Index subclassing Int64Index. its not a type (its dtype is int64).  types can be the dtype of an Index.